pkg/notifier/shared.go

```diff
@@ -0,0 +1,89 @@
+package notifier
+
+import (
+	context "context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/hashicorp/go-plugin"
+	grpc "google.golang.org/grpc"
+)
+
+type Notifier interface {
```

Comment:
why are not you using notifier.NotifierServer? (the one autogenerated from proto)